{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "d6938e90-7a24-4b9a-a795-314e9714a42b",
   "metadata": {},
   "source": [
    "# SMR\n",
    "- Based on SMR_template.txt\n",
    "- run updated v8 GTEx on all Summary Statistics (including new BD summary statistics)\n",
    "- automate swarm creation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5c8f80bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "#REF first two 'hits' upon googling for \"where should i install modules for use with jupyter notebook\" as at 2023-11-22 15:27GMT (HD):\n",
    "#\n",
    "#!pip install pandas   #<-- Commented this line as at 2023-11-23 20:50GMT (HD)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "25b21697-7612-45a1-83e7-e4ae100f17cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "import zipfile\n",
    "import pandas as pd\n",
    "import os\n",
    "import glob\n",
    "import numpy as np\n",
    "import scipy\n",
    "from scipy.stats import norm\n",
    "import sys\n",
    "import subprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "17253793-86a8-4c15-9652-b687062fe3fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "def shell_do(command, log=False, return_log=False, make_part=False):\n",
    "    print(f'Executing: {(\" \").join(command.split())}', file=sys.stderr)\n",
    "\n",
    "    if make_part == False:\n",
    "        res = subprocess.run(command.split(), stdout=subprocess.PIPE)\n",
    "    else:\n",
    "        res = subprocess.run(command, shell=True, stdout=subprocess.PIPE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "30e1a2c9-4d3e-4254-a77c-13f7895808e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "def SMR(disease, stats_path, work_dir, out_dir):\n",
    "    ### START Code temporarily(?) commented as at 2023-12-19 10:35GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    ##with open(f'.../omicSynth/v8/swarms/{disease}.SMR_swarm', 'w') as f:   #<-- Commented this line as at 2023-11-23 20:28GMT (HD).\n",
    "    ##\n",
    "    ##v Line below 'new' code as at 2023-11-23 20:28GMT (HD). v\n",
    "    #with open(f'/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/swarms/{disease}.SMR_swarm', 'w') as f:\n",
    "    #    #\n",
    "    #    #\n",
    "    #    #f.write('cd .../omicSynth/v8/intermediate_results/ \\n')   #<-- Commented this line as at 2023-11-23 20:23GMT (HD).\n",
    "    #    #\n",
    "    #    f.write('cd \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/intermediate_results/\" \\n')   #<-- This line 'new' code as at 2023-11-23 20:32GMT (HD).\n",
    "    #    #\n",
    "    #    #\n",
    "    #    #\n",
    "    #    #\n",
    "    ### END Code temporarily(?) commented as at 2023-12-19 10:35GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    ### START 'new' code as at 2023-12-19 10:33GMT (HD)\n",
    "    #\n",
    "    #\n",
    "    cmd = f'cd \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/intermediate_results/\"'\n",
    "    \n",
    "    shell_do(cmd, make_part=True)\n",
    "    #\n",
    "    #\n",
    "    ### END 'new' code as at 2023-12-19 10:33GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    ### START Code temporarily(?) commented as at 2023-12-19 10:37GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #f.write('module load SMR \\n')\n",
    "    #f.write('\\n')\n",
    "    #f.write('\\n')\n",
    "    #f.write('# Single file per assay starts here. \\n')\n",
    "    #f.write('\\n')\n",
    "    #f.write('## expression_blood_eQTLgen ## \\n')\n",
    "    #f.write(f'smr --gwas-summary {stats_path} --beqtl-summary {work_dir}/SMR_omics/eQTLs/eQTLgen/cis-eQTLs-full_eQTLGen_AF_incl_nr_formatted_20191212.new.txt_besd-dense --ld-upper-limit --out {out_dir}/{disease}_expression_blood_eQTLgen_SMR_allChrs --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel --smr-multi \\n')\n",
    "    #f.write(' \\n')\n",
    "    #f.write('## expression_brain_eMeta ## \\n')\n",
    "    ##\n",
    "    ##\n",
    "    ##f.write(f'smr --gwas-summary {stats_path} --beqtl-summary {work_dir}/SMR_omics/eQTLs/Brain-eMeta/Brain-eMeta --ld-upper-limit --out {out_dir}/{disease}_expression_brain_eMeta_SMR_allChrs --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel --smr-multi \\n')   #<-- Commented this line as at 2023-11-23 20:35GMT (HD), and uncommented again as at 2023-12-14 13:43GMT (HD):, and commented again as at 2023-12-19 09:51GMT (HD).\n",
    "    ##Googling for \"brain-emeta\" yielded as one of the 'hits' \"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8053199/\":\n",
    "    ##====\n",
    "    ##In addition, we repeated the analysis by using the meta-analyzed version of the eQTL summary data (named brain-eMeta hereafter), which included results from the GTEx data of brain tissues22, the Common Mind Consortium23, and the Religious Orders Study and the Rush Memory and Aging Project24. Results from these three studies were meta-analyzed using the MeCS method (meta-analysis of cis-eQTL in correlated samples) to increase the power of detecting brain eQTLs25. Only SNPs within 1â€‰Mb distance from each individual probe are available. The eQTL data can be downloaded at https://cnsgenomics.com/data/SMR/#eQTLsummarydata\n",
    "    ##====\n",
    "    ##==> \"https://cnsgenomics.com/data/SMR/#eQTLsummarydata\" includes \"\tBrain-eMeta.tar.gz\" (== \"https://cnsgenomics.com/data/SMR/Brain-eMeta.tar.gz\"), so could continue with downloading that and using *that* as a (proper?) reference for \"Brain-eMeta\"???? (!!)\n",
    "    ##HOWEVER, also, \"https://yanglab.westlake.edu.cn/software/smr/#DataResource\" mentions that:\n",
    "    ##====\n",
    "    ##Brain-eMeta ([* -- HD for emphasis]now called BrainMeta v1[* -- HD end for emphasis]) eQTL data (estimated effective n = 1194)\n",
    "    ##\n",
    "    ##BrarinMeta[sic -- HD] eQTL summary data (Qi et al. 2018 Nat Commun) in SMR binary (BESD) format: BrainMeta_v1.tar.gz (1.1 GB)\n",
    "    ##\n",
    "    ##This is a set of eQTL data from a meta-analysis of GTEx brain (GTEx Consortium 2017 Nature), CMC (Fromer et al. 2016 Nat Neurosci), and ROSMAP (Ng et al. 2017 Nat Neurosci) by MeCS. Only SNPs within 1Mb distance from each probe are available.\n",
    "    ##====\n",
    "    ##... so \"BrainMeta[ v1]\" should actually equate to \"Brain-eMeta\"... grmbl i.e. not necessarily making matters easier IMHO 'but OK'.\n",
    "    ##\n",
    "    ##f.write(f'smr --gwas-summary {stats_path} --beqtl-summary {work_dir}/SMR_omics/eQTLs/Brain-eMeta/Brain-eMeta --ld-upper-limit --out {out_dir}/{disease}_expression_brain_eMeta_SMR_allChrs --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel\" --smr-multi \\n')   #<-- This line 'new' code as at 2023-11-24 21:21GMT (HD) -- + NOTE: *ASSUMING* that Alvarado et al. medRxiv-2023 considered *cis*-eQTLs to be Brain-eMeta eQTLs: REF \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/SMR_omics/eQTLs/Brain-eMeta/Brain-eMeta\" (i.e. a symbolic link to a directory holding \"BrainMeta_cis_eqtl_summary\" data [sic], as downloaded from \"https://yanglab.westlake.edu.cn/data/SMR\"). #Commented this line as at 2023-12-11 16:55GMT (HD), as I obtained the below ERROR:\n",
    "    ##====\n",
    "    ##(base) hammie@hammie-Default-string:/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr$ smr-1.3.1 --gwas-summary '/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariants_mygwasDOTmaFormatFieldsOnly.tab' --beqtl-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/SMR_omics/eQTLs/Brain-eMeta/Brain-eMeta\" --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_allChrs\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel\" --smr-multi\n",
    "    ##*******************************************************************\n",
    "    ##* Summary-data-based Mendelian Randomization (SMR)\n",
    "    ##* Version 1.3.1\n",
    "    ##* Build at Sep 21 2022 12:13:19, by GCC 8.3\n",
    "    ##* (C) 2015 Futao Zhang, Zhihong Zhu and Jian Yang\n",
    "    ##* The University of Queensland\n",
    "    ##* MIT License\n",
    "    ##*******************************************************************\n",
    "    ##Analysis started: 16:29:13,Mon Dec 11,2023\n",
    "    ##\n",
    "    ##Options:\n",
    "    ##--gwas-summary /media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariants_mygwasDOTmaFormatFieldsOnly.tab\n",
    "    ##--beqtl-summary /media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/SMR_omics/eQTLs/Brain-eMeta/Brain-eMeta\n",
    "    ##--out /media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_allChrs\n",
    "    ##--bfile /media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel\n",
    "    ##--smr-multi \n",
    "    ##\n",
    "    ##Reading GWAS summary data from [/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariants_mygwasDOTmaFormatFieldsOnly.tab].\n",
    "    ##GWAS summary data of 17510617 SNPs to be included from [/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariants_mygwasDOTmaFormatFieldsOnly.tab].\n",
    "    ##\n",
    "    ##ERROR: can not open the file [/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/SMR_omics/eQTLs/Brain-eMeta/Brain-eMeta.esi] to read.\n",
    "    ##\n",
    "    ##Analysis completed: 16:29:58,Mon Dec 11,2023\n",
    "    ##Computational time: 0:0:45\n",
    "    ##(base) hammie@hammie-Default-string:/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr$ pwd\n",
    "    ##/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr\n",
    "    ##(base) hammie@hammie-Default-string:/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr$\n",
    "    ##====\n",
    "    ##\n",
    "    ##f.write(f'smr --gwas-summary {stats_path} --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr1 --ld-upper-limit --out {out_dir}/{disease}_expression_brain_eMeta_SMR_allChrs --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel\" --smr-multi \\n')   #<-- This line 'new' code as at 2023-12-11 17:02GMT (HD) -- + NOTE: *ASSUMING* that Alvarado et al. medRxiv-2023 considered *cis*-eQTLs to be Brain-eMeta eQTLs: REF \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/SMR_omics/eQTLs/Brain-eMeta/Brain-eMeta\" (i.e. a symbolic link to a directory holding \"BrainMeta_cis_eqtl_summary\" data [sic], as downloaded from \"https://yanglab.westlake.edu.cn/data/SMR\"). #Commented this line as at 2023-12-11 17:22GMT (HD), as its execution yielded the following ERROR:\n",
    "    ##====\n",
    "    ##(base) hammie@hammie-Default-string:/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr$ smr-1.3.1 --gwas-summary '/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariants_mygwasDOTmaFormatFieldsOnly.tab' --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr1 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_allChrs\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel\" --smr-multi\n",
    "    ##*******************************************************************\n",
    "    ##* Summary-data-based Mendelian Randomization (SMR)\n",
    "    ##* Version 1.3.1\n",
    "    ##* Build at Sep 21 2022 12:13:19, by GCC 8.3\n",
    "    ##* (C) 2015 Futao Zhang, Zhihong Zhu and Jian Yang\n",
    "    ##* The University of Queensland\n",
    "    ##* MIT License\n",
    "    ##*******************************************************************\n",
    "    ##Analysis started: 17:6:34,Mon Dec 11,2023\n",
    "    ##\n",
    "    ##Options:\n",
    "    ##--gwas-summary /media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariants_mygwasDOTmaFormatFieldsOnly.tab\n",
    "    ##--beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr1\n",
    "    ##--out /media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_allChrs\n",
    "    ##--bfile /media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel\n",
    "    ##--smr-multi \n",
    "    ##\n",
    "    ##Reading GWAS summary data from [/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariants_mygwasDOTmaFormatFieldsOnly.tab].\n",
    "    ##GWAS summary data of 17510617 SNPs to be included from [/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariants_mygwasDOTmaFormatFieldsOnly.tab].\n",
    "    ##Reading eQTL SNP information from [/home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr1.esi].\n",
    "    ##920092 SNPs to be included from [/home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr1.esi].\n",
    "    ##\n",
    "    ##Error: can not open the file [/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel.fam] to read.\n",
    "    ##\n",
    "    ##Analysis completed: 17:7:24,Mon Dec 11,2023\n",
    "    ##(base) hammie@hammie-Default-string:/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr(base) hammie@hammie-Default-string:/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr$ pwd\n",
    "    ##/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr\n",
    "    ##(base) hammie@hammie-Default-string:/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr$       \n",
    "    ##====\n",
    "    ##\n",
    "    ##f.write(f'smr --gwas-summary {stats_path} --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr1 --ld-upper-limit --out {out_dir}/{disease}_expression_brain_eMeta_SMR_allChrs --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi \\n')   #<-- This line 'new' code as at 2023-12-11 17:24GMT (HD) -- + NOTE: *ASSUMING* that Alvarado et al. medRxiv-2023 considered *cis*-eQTLs to be Brain-eMeta eQTLs: REF \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/SMR_omics/eQTLs/Brain-eMeta/Brain-eMeta\" (i.e. a symbolic link to a directory holding \"BrainMeta_cis_eqtl_summary\" data [sic], as downloaded from \"https://yanglab.westlake.edu.cn/data/SMR\"). #Commented this line as at 2023-12-19 09:59GMT (HD).\n",
    "    ##\n",
    "    ##\n",
    "    ##        \n",
    "    ### END Code temporarily(?) commented as at 2023-12-19 10:37GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    ### START 'new' code as at 2023-12-19 09:57GMT (HD)\n",
    "    #\n",
    "    #\n",
    "    ### START Code commented as at 2023-12-21 12:36GMT (HD), as superfluous by now\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #cmd = f'PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"'\n",
    "    #\n",
    "    #shell_do(cmd, make_part=True)\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    ### END Code commented as at 2023-12-21 12:36GMT (HD).\n",
    "    #\n",
    "    #v Note re below: 'only' 22 chromosomes, as per contents of \"/home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary\" as on the Brutus as at 2023-12-19 10:04GMT (HD). v\n",
    "    #\n",
    "    #v TEMPorarily commented line below, for testing only as at 2023-12-21 11:26GMT (HD). v #Uncommented this line again as at 2023-12-21 12:52GMT (HD).\n",
    "    #for i in range(1,23):\n",
    "    #\n",
    "    #v Line below 'new' and temporary, for testing purposes (i.e., run analysis for CHR 22 *only* -- note that Python's range() function counts up to but not including the second number mentioned) only as at 2023-12-21 11:29GMT (HD). v\n",
    "    for i in range(22,23):   #<-- Commented this line as at 2023-12-21 12:51GMT (HD).\n",
    "        #cmd = f'smr --gwas-summary {stats_path} --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr{i} --ld-upper-limit --out {out_dir}/{disease}_expression_brain_eMeta_SMR_chr{i} --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi'   #<-- This line 'new' code as at 2023-12-11 17:24GMT (HD) -- + NOTE: *ASSUMING* that Alvarado et al. medRxiv-2023 considered *cis*-eQTLs to be Brain-eMeta eQTLs: REF \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/SMR_omics/eQTLs/Brain-eMeta/Brain-eMeta\" (i.e. a symbolic link to a directory holding \"BrainMeta_cis_eqtl_summary\" data [sic], as downloaded from \"https://yanglab.westlake.edu.cn/data/SMR\"). #Commented this line as at 2023-12-19 17:53GMT (HD).\n",
    "        #\n",
    "        #cmd = f'smr-1.3.1 --gwas-summary {stats_path} --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr{i} --ld-upper-limit --out {out_dir}/{disease}_expression_brain_eMeta_SMR_chr{i} --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi'   #<-- This line 'new' code as at 2023-12-11 17:24GMT (HD) -- + NOTE: *ASSUMING* that Alvarado et al. medRxiv-2023 considered *cis*-eQTLs to be Brain-eMeta eQTLs: REF \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/SMR_omics/eQTLs/Brain-eMeta/Brain-eMeta\" (i.e. a symbolic link to a directory holding \"BrainMeta_cis_eqtl_summary\" data [sic], as downloaded from \"https://yanglab.westlake.edu.cn/data/SMR\"). #This line 'new' code as at 2023-12-19 17:55GMT (HD), as the following worked in an OS terminal on the Brutus: #<-- Commented this line as at 2023-12-19 18:02GMT (HD), as it produced an error even though issuing \"$ smr-1.3.1\" in an OS terminal on the Brutus worked. #... and uncommented this line again as at 2023-12-19 18:21GMT (HD)., and recommented it again as at 2023-12-19 18:26GMT (HD).\n",
    "        #====\n",
    "        #(base) hammie@hammie-Default-string:/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr$ smr-1.3.1 --gwas-summary '/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab' --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr1 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_allChrs\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "        #====\n",
    "        #\n",
    "        #cmd = f'PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary {stats_path} --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr{i} --ld-upper-limit --out {out_dir}/{disease}_expression_brain_eMeta_SMR_chr{i} --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi'   #<-- This line 'new' code as at 2023-12-11 17:24GMT (HD) -- + NOTE: *ASSUMING* that Alvarado et al. medRxiv-2023 considered *cis*-eQTLs to be Brain-eMeta eQTLs: REF \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/SMR_omics/eQTLs/Brain-eMeta/Brain-eMeta\" (i.e. a symbolic link to a directory holding \"BrainMeta_cis_eqtl_summary\" data [sic], as downloaded from \"https://yanglab.westlake.edu.cn/data/SMR\"). #This line 'new' code as at 2023-12-19 18:05GMT (HD) #Commented this line as at 2023-12-19 18:20GMT (HD) -- however, it appears to be necessary as when I issued the \"$PATH\" amendment only once, as at 2023-12-19 18:23GMT I obtained the following errors: #Commented this line as at 2023-12-21 11:40GMT (HD).\n",
    "        #====\n",
    "        #Executing: cd \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/intermediate_results/\"\n",
    "        #Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"\n",
    "        #Executing: smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariants_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr1 --ld-upper-limit --out /media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr1 --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "        #/bin/sh: 1: smr-1.3.1: not found\n",
    "        #Executing: smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariants_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr2 --ld-upper-limit --out /media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr2 --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "        #/bin/sh: 1: smr-1.3.1: not found\n",
    "        #Executing: smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariants_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr3 --ld-upper-limit --out /media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr3 --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "        #/bin/sh: 1: smr-1.3.1: not found\n",
    "        #[... -- HD]\n",
    "        #====\n",
    "        #\n",
    "        cmd = f'PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary {stats_path} --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr{i} --out \"{out_dir}/{disease}_expression_brain_eMeta_SMR_chr{i}\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi'   #<-- This line 'new' code as at 2023-12-21 11:40GMT (HD), as I had been obtaining the below error:\n",
    "        #====\n",
    "        #(base) hammie@hammie-Default-string:/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr$ pwd\n",
    "        #/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr\n",
    "        #(base) hammie@hammie-Default-string:/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr$ smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariants_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr22 --ld-upper-limit --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test\"/PD_expression_brain_eMeta_SMR_chr22TESTONLY --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "        #*******************************************************************\n",
    "        #* Summary-data-based Mendelian Randomization (SMR)\n",
    "        #* Version 1.3.1\n",
    "        #* Build at Sep 21 2022 12:13:19, by GCC 8.3\n",
    "        #* (C) 2015 Futao Zhang, Zhihong Zhu and Jian Yang\n",
    "        #* The University of Queensland\n",
    "        #* MIT License\n",
    "        #*******************************************************************\n",
    "        #Analysis started: 11:33:17,Thu Dec 21,2023\n",
    "        #\n",
    "        #Options:\n",
    "        #--gwas-summary /media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariants_mygwasDOTmaFormatFieldsOnly.tab\n",
    "        #--beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr22\n",
    "        #--ld-upper-limit 0.000000\n",
    "        #--bfile /media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\n",
    "        #--smr-multi \n",
    "        #\n",
    "        #ERROR: --ld-min 0.050000 is larger than --ld-top 0.000000.\n",
    "        #(base) hammie@hammie-Default-string:/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr$\n",
    "        #====\n",
    "        \n",
    "        shell_do(cmd, make_part=True)\n",
    "    #f.write(' \\n')        \n",
    "    #\n",
    "    #\n",
    "    ### END 'new' code as at 2023-12-19 09:57GMT (HD).\n",
    "    #        \n",
    "    #\n",
    "    #\n",
    "    ### START Code temporarily(?) commented as at 2023-12-19 17:14GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #f.write(' \\n')\n",
    "    #f.write('## methylation_brain_mMeta ## \\n')\n",
    "    #f.write(f'smr --gwas-summary {stats_path} --beqtl-summary {work_dir}/SMR_omics/regQTLs/Brain-mMeta/Brain-mMeta --ld-upper-limit --out {out_dir}/{disease}_methylation_brain_mMeta_SMR_allChrs --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel --smr-multi \\n')\n",
    "    #f.write('## chromatin_blood_Bryois ## \\n')\n",
    "    #f.write(f'smr --gwas-summary {stats_path} --beqtl-summary {work_dir}/SMR_omics/regQTLs/Bryois_caQTLs/bryois_NatCommun_2018_50kb_cQTLs --ld-upper-limit --out {out_dir}/{disease}_chromatin_blood_Bryois_SMR_allChrs --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel --smr-multi \\n')\n",
    "    #f.write(' \\n')\n",
    "    #f.write('# Multi chromosome reference data starts here. \\n')\n",
    "    #f.write(' \\n')\n",
    "    # \n",
    "    #f.write('## methylation_blood_mcrae ## \\n')\n",
    "    #for i in range(1,23): \n",
    "    #    f.write(f'smr --gwas-summary {stats_path} --beqtl-summary {work_dir}/SMR_omics/regQTLs/Blood-mMeta/bl_mqtl_lite_chr{i} --out {out_dir}/{disease}_methylation_blood_mcrae_SMR_chr{i} --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel --smr-multi \\n')\n",
    "    #f.close()\n",
    "    #\n",
    "    ## swarm command\n",
    "    ##cmd = f'swarm -b 15 -f .../omicSynth/v8/swarms/{disease}.SMR_swarm -g 48 -t 12 --logdir .../omicSynth/v8/swarms --module SMR'   #<-- Commented this line as at 2023-11-23 20:49GMT (HD).\n",
    "    ##\n",
    "    #cmd = f'swarm -b 15 -f \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/swarms/{disease}.SMR_swarm\" -g 48 -t 12 --logdir \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/swarms\" --module SMR'   #<-- This line 'new' code as at 2023-11-23 20:49GMT (HD).\n",
    "    #\n",
    "    #shell_do(cmd)\n",
    "    #\n",
    "    #\n",
    "    ### END Code temporarily(?) commented as at 2023-12-19 17:14GMT (HD).\n",
    "\n",
    "def SMR_meta(disease, stats_path, work_dir, out_dir):\n",
    "    with open(f'.../omicSynth/v8/swarms/{disease}_meta.SMR_swarm', 'w') as f:\n",
    "        f.write('cd .../omicSynth/intermediate_results/ \\n')\n",
    "        f.write('module load SMR \\n')\n",
    "        f.write('\\n')\n",
    "        f.write('\\n')\n",
    "        f.write('# Multi chromosome reference data starts here. \\n')\n",
    "        f.write(' \\n')\n",
    "        \n",
    "        f.write('## expression_basalganglia_metaBrain ## \\n')\n",
    "        for i in range(1,23):\n",
    "            f.write(f'smr --gwas-summary {stats_path} --beqtl-summary {work_dir}/SMR_omics/eQTLs/2020-05-26-Basalganglia-EUR/2020-05-26-Basalganglia-EUR-{i}-SMR-besd --out {out_dir}/{disease}_expression_Basalganglia_metaBrain_SMR_chr{i} --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel_gr38_metaBrain_variant_names --smr-multi \\n')\n",
    "        f.write(' \\n')\n",
    "\n",
    "        f.write('## expression_cerebellum_metaBrain ## \\n')\n",
    "        for i in range(1,23):\n",
    "            f.write(f'smr --gwas-summary {stats_path} --beqtl-summary {work_dir}/SMR_omics/eQTLs/2020-05-26-Cerebellum-EUR/2020-05-26-Cerebellum-EUR-{i}-SMR-besd --out {out_dir}/{disease}_expression_Cerebellum_metaBrain_SMR_chr{i} --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel_gr38_metaBrain_variant_names --smr-multi \\n')\n",
    "        f.write(' \\n')\n",
    "\n",
    "        f.write('## expression_cortex_metaBrain ## \\n')\n",
    "        for i in range(1,23):    \n",
    "            f.write(f'smr --gwas-summary {stats_path} --beqtl-summary {work_dir}/SMR_omics/eQTLs/2020-05-26-Cortex-EUR/2020-05-26-Cortex-EUR-{i}-SMR-besd --out {out_dir}/{disease}_expression_Cortex_metaBrain_SMR_chr{i} --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel_gr38_metaBrain_variant_names --smr-multi \\n')\n",
    "        f.write(' \\n')\n",
    "\n",
    "        f.write('## expression_hippocampus_metaBrain ## \\n')\n",
    "        for i in range(1,23):  \n",
    "            f.write(f'smr --gwas-summary {stats_path} --beqtl-summary {work_dir}/SMR_omics/eQTLs/2020-05-26-Hippocampus-EUR/2020-05-26-Hippocampus-EUR-{i}-SMR-besd --out {out_dir}/{disease}_expression_Hippocampus_metaBrain_SMR_chr{i} --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel_gr38_metaBrain_variant_names --smr-multi \\n')\n",
    "        f.write(' \\n')\n",
    "\n",
    "        f.write('## expression_spinalchord_metaBrain ## \\n')\n",
    "        for i in range(1,23): \n",
    "            f.write(f'smr --gwas-summary {stats_path} --beqtl-summary {work_dir}/SMR_omics/eQTLs/2020-05-26-Spinalcord-EUR/2020-05-26-Spinalcord-EUR-{i}-SMR-besd --out {out_dir}/{disease}_expression_Spinalcord_metaBrain_SMR_chr{i} --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel_gr38_metaBrain_variant_names --smr-multi \\n')\n",
    "        f.write(' \\n')\n",
    "        f.close()\n",
    "        \n",
    "    # swarm command\n",
    "    cmd = f'swarm -b 15 -f .../omicSynth/v8/swarms/{disease}_meta.SMR_swarm -g 48 -t 12 --logdir .../omicSynth/v8/swarms --module SMR'\n",
    "\n",
    "    shell_do(cmd)\n",
    "\n",
    "def SMR_gtex(disease, stats_path, work_dir, out_dir, gtex_list):\n",
    "    ### For this function definition, try and do analogously as for SMR()'s definition above: START Code temporarily(?) commented as at 2023-12-21 17:10GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #with open(f'.../omicSynth/v8/swarms/{disease}_gtex.SMR_swarm', 'w') as f:\n",
    "    #    f.write('cd .../micSynth/v8/intermediate_results/ \\n')\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    ### END Code temporarily(?) commented as at 2023-12-21 17:10GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    ### START code copy--pasted from SMR() function definition above as at 2023-12-21 17:13GMT (HD)\n",
    "    #\n",
    "    #\n",
    "    cmd = f'cd \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/intermediate_results/\"'\n",
    "    \n",
    "    shell_do(cmd, make_part=True)\n",
    "    #\n",
    "    #\n",
    "    ### END code copy--pasted from SMR() function definition above as at 2023-12-21 17:13GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    ### START Code temporarily(?) commented as at 2023-12-21 17:20GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #f.write('module load SMR \\n')\n",
    "    #f.write('\\n')\n",
    "    #f.write('\\n')\n",
    "    #f.write('# expression_GTEx \\n')\n",
    "    #f.write(' \\n')\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    ### END Code temporarily(?) commented as at 2023-12-21 17:20GMT (HD).\n",
    "\n",
    "    for part in gtex_list:   #<-- TEMPorarily commented this line, for testing purposes only as at 2023-12-22 20:57GMT (HD)\n",
    "    #\n",
    "    #v Line below 'new' code, for testing only (this way, would analyses be run for first tissue listed in 'gtexlist' only indeed? <-- A: NO -- as at 2023-12-22 22:25GMT [HD]: i.e. this simply doesn't work) as at 2023-12-22 20:58GMT (HD) v\n",
    "    #for part in gtex_list[1]:\n",
    "        #f.write(f'smr --gwas-summary {stats_path} --beqtl-summary .../omicSynth/SMR_omics/eQTLs/GTEx_v8/eQTL_besd_lite/{part} --out {out_dir}/{disease}_expression_{part}_GTEx_SMR_allChrs --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel  --smr-multi \\n')   #<-- Commented this line as at 2023-12-21 17:27GMT (HD).\n",
    "        \n",
    "        ### START code amended from SMR() function definition above as at 2023-12-21 17:29GMT (HD).\n",
    "        #\n",
    "        #\n",
    "        cmd = f'PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary {stats_path} --beqtl-summary /home/hammie/DATA/SMR/GTEx_V8_cis_eqtl_summary_lite/eQTL_besd_lite/{part} --out \"{out_dir}/{disease}_expression_{part}_GTEx_SMR_allChrs\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi'\n",
    "        \n",
    "        shell_do(cmd, make_part=True)\n",
    "        #\n",
    "        #\n",
    "        ### END code amended from SMR() function definition above as at 2023-12-21 17:29GMT (HD).\n",
    "        \n",
    "    ### START Code temporarily(?) commented as at 2023-12-21 17:24GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #f.close()\n",
    "    #\n",
    "    ## swarm command\n",
    "    #cmd = f'swarm -b 15 -f .../omicSynth/v8/swarms/{disease}_gtex.SMR_swarm -g 48 -t 12 --logdir .../omicSynth/v8/swarms --module SMR'\n",
    "    #\n",
    "    #shell_do(cmd)\n",
    "    #\n",
    "    #\n",
    "    ### END Code temporarily(?) commented as at 2023-12-21 17:24GMT (HD).\n",
    "\n",
    "def SMR_psych(disease, stats_path, work_dir, out_dir):\n",
    "    with open(f'.../omicSynth/v8/swarms/{disease}_psych.SMR_swarm', 'w') as f:\n",
    "        f.write('module load SMR \\n')\n",
    "        f.write('\\n')\n",
    "        f.write('\\n')\n",
    "        f.write('# Single file per assay starts here. \\n')\n",
    "        f.write('\\n')\n",
    "        f.write('## expression_psychencode prefrontal cortex ## \\n')\n",
    "        f.write(f'smr --gwas-summary {stats_path} --beqtl-summary .../omicSynth/SMR_omics/eQTLs/PsychEncode/DER-08a_hg19_eQTL.significant --out {out_dir}/{disease}_expression_psychEncode_prefrontal_cortex_SMR_allChrs --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel --smr-multi \\n')\n",
    "        f.write(' \\n')\n",
    "    \n",
    "    # swarm command\n",
    "    cmd = f'swarm -b 20 -f ...//omicSynth/v8/swarms/{disease}_psych.SMR_swarm -g 48 --logdir .../omicSynth/v8/swarms --time=12:00:00 --module SMR'\n",
    "    \n",
    "    shell_do(cmd)\n",
    "\n",
    "def SMR_pQTL(disease, stats_path, work_dir, out_dir):\n",
    "    with open(f'.../omicSynth/v8/swarms/{disease}_atlas.SMR_swarm', 'w') as f:\n",
    "        f.write('module load SMR \\n')\n",
    "        f.write('\\n')\n",
    "        f.write('\\n')\n",
    "        f.write('# Single file per assay starts here. \\n')\n",
    "        f.write('\\n')\n",
    "        f.write('## pQTL CSF ## \\n')\n",
    "        f.write(f'smr --gwas-summary {stats_path} --beqtl-summary .../omicSynth/SMR_omics/pQTLs/atlas/csf_atlas_update --out {out_dir}/{disease}_pQTL_csf_SMR_allChrs --bfile .../additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel --smr-multi \\n')\n",
    "        f.write(' \\n')\n",
    "        f.write('## pQTL plasma ## \\n')\n",
    "        f.write(f'smr --gwas-summary {stats_path} --beqtl-summary .../omicSynth/SMR_omics/pQTLs/atlas/plasma_atlas_update --out {out_dir}/{disease}_pQTL_plasma_SMR_allChrs --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel --smr-multi \\n')\n",
    "        f.write(' \\n')\n",
    "        f.write('## pQTL brain ## \\n')\n",
    "        f.write(f'smr --gwas-summary {stats_path} --beqtl-summary .../omicSynth/SMR_omics/pQTLs/atlas/brain_atlas_update --out {out_dir}/{disease}_pQTL_brain_SMR_allChrs --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel --smr-multi \\n')\n",
    "        f.write(' \\n')\n",
    "    \n",
    "    # swarm command\n",
    "    cmd = f'swarm -b 20 -f .../omicSynth/v8/swarms/{disease}_atlas.SMR_swarm -g 48 --logdir .../omicSynth/v8/swarms --time=12:00:00 --module SMR'\n",
    "    \n",
    "    shell_do(cmd)\n",
    "\n",
    "def SMR_eQTL(disease, stats_path, work_dir, out_dir):\n",
    "    with open(f'.../omicSynth/v8/swarms/{disease}_multiancestry.SMR_swarm', 'w') as f:\n",
    "        f.write('module load SMR \\n')\n",
    "        f.write('\\n')\n",
    "        f.write('\\n')\n",
    "        f.write('# Single file per assay starts here. \\n')\n",
    "        f.write('\\n')\n",
    "        f.write('## eQTL multi ancestry ## \\n')\n",
    "        f.write(f'smr --gwas-summary {stats_path} --beqtl-summary .../omicSynth/SMR_omics/eQTLs/multiancestry/multi_ancestry_eqtl --out {out_dir}/{disease}_eQTL_multiancestry_SMR_allChrs --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel --smr-multi \\n')\n",
    "        f.write(' \\n')\n",
    "\n",
    "    # swarm command\n",
    "    cmd = f'swarm -b 20 -f .../omicSynth/v8/swarms/{disease}_multiancestry.SMR_swarm -g 48 --logdir .../omicSynth/v8/swarms --time=12:00:00 --module SMR'\n",
    "    \n",
    "    shell_do(cmd)\n",
    "\n",
    "def SMR_eqtlGen(disease, stats_path, work_dir, out_dir):\n",
    "    ### For this function definition, try and do analogously as for SMR()'s and SMR_gtex()'s' definitions above: START Code temporarily(?) commented as at 2023-12-22 22:02GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #with open(f'.../omicSynth/v8/swarms/{disease}_eqtlgen.SMR_swarm', 'w') as f:\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    ### END Code temporarily(?) commented as at 2023-12-22 22:02GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    ### START code copy--pasted from SMR() function definition above as at 2023-12-22 22:07GMT (HD)\n",
    "    #\n",
    "    #\n",
    "    cmd = f'cd \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/intermediate_results/\"'\n",
    "    \n",
    "    shell_do(cmd, make_part=True)\n",
    "    #\n",
    "    #\n",
    "    ### END code copy--pasted from SMR() function definition above as at 2023-12-22 22:07GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    ### START Code temporarily(?) commented as at 2023-12-22 22:08GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #f.write('module load SMR \\n')\n",
    "    #f.write('\\n')\n",
    "    #f.write('\\n')\n",
    "    #f.write('# Single file per assay starts here. \\n')\n",
    "    #f.write('\\n')\n",
    "    #f.write('## expression_blood_eQTLgen ## \\n')\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    ### END Code temporarily(?) commented as at 2023-12-22 22:08GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #f.write(f'smr --gwas-summary {stats_path} --beqtl-summary {work_dir}/SMR_omics/eQTLs/eQTLgen/cis-eQTLs-full_eQTLGen_AF_incl_nr_formatted_20191212.new.txt_besd-dense --out {out_dir}/{disease}_expression_blood_eQTLgen_SMR_allChrs --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel --smr-multi \\n')   #<-- Commented this line as at 2023-12-22 21:49GMT (HD).\n",
    "    #\n",
    "    ### START 'new' code as at 2023-12-22 21:49GMT (HD).\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    cmd = f'PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary {stats_path} --beqtl-summary \"/media/hammie/My Passport/DATA/eQTLGen/cis-eQTLs-full_eQTLGen_AF_incl_nr_formatted_20191212.new.txt_besd-dense\" --out \"{out_dir}/{disease}_expression_blood_eQTLgen_SMR_allChrs\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi'\n",
    "    \n",
    "    shell_do(cmd, make_part=True)\n",
    "    #\n",
    "    #\n",
    "    #\n",
    "    ### END 'new' code as at 2023-12-22 21:49GMT (HD).\n",
    "        \n",
    "    ### START Code temporarily(?) commented as at 2023-12-22 22:12GMT (HD).\n",
    "    #    #\n",
    "    #    #\n",
    "    #    f.write(' \\n')\n",
    "    #    f.close()\n",
    "    #\n",
    "    ## swarm command\n",
    "    #cmd = f'swarm -b 15 -f .../omicSynth/v8/swarms/{disease}_eqtlgen.SMR_swarm -g 48 -t 12 --logdir .../omicSynth/v8/swarms --module SMR'\n",
    "    #\n",
    "    #shell_do(cmd)\n",
    "    #\n",
    "    #\n",
    "    ### END Code temporarily(?) commented as at 2023-12-22 22:12GMT (HD)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "9f908c47-5a9f-44cf-b440-6605e511b8b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "#v8_GTEx_path = '.../omicSynth/SMR_omics/eQTLs/GTEx_v8/eQTL_besd_lite/'   #<-- Commented this line as at 2023-11-23 20:17GMT (HD).\n",
    "#\n",
    "v8_GTEx_path = '/home/hammie/DATA/SMR/GTEx_V8_cis_eqtl_summary_lite/eQTL_besd_lite/'   #<-- This line 'new' code as at 2023-11-23 20:18GMT (HD).\n",
    "\n",
    "# define remaining paths\n",
    "#\n",
    "#\n",
    "#\n",
    "#out_dir = '.../omicSynth/v8/intermediate_results'   #<-- Commented this line as at 2023-11-23 19:43GMT (HD).\n",
    "#\n",
    "out_dir = '/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/intermediate_results'   #<-- This line 'new' code as at 2023-11-23 19:43GMT (HD).\n",
    "#\n",
    "#\n",
    "# work_dir = '.../omicSynth'   #<-- Commented this line as at 2023-11-23 18:28GMT (HD).\n",
    "#\n",
    "work_dir = '/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis'   #<-- This line 'new' code as at 2023-11-23 19:27GMT (HD)."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a2728cb2-ef58-45da-a445-dd93de07bd80",
   "metadata": {},
   "source": [
    "## All Diseases"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c130f755-4077-46ba-b6ec-23894f70d567",
   "metadata": {},
   "source": [
    "### Data Prep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "83e9eb41-416b-4ea9-a2df-94e3edccca32",
   "metadata": {},
   "outputs": [],
   "source": [
    "# pull all GTEx paths\n",
    "gtex_raw = glob.glob(f'{v8_GTEx_path}/*') \n",
    "\n",
    "gtex_short = [] # list to hold gtex tissue names\n",
    "for x in gtex_raw:\n",
    "    short_dx = x.split('/')[-1].rsplit('.',1)[0] # strip path to obtain the tissue name\n",
    "    if short_dx not in gtex_short:\n",
    "        gtex_short.append(short_dx)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "f5ad824d-e1cb-4073-9451-63398d8d3fd6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#dx_list = glob.glob('.../omicSynth/outcome_summary_stats/*.ma') # paths so we can list out names of diseases we have sum stats for   #<-- Commented this line as at 2023-12-19 19:04GMT (HD).\n",
    "#\n",
    "dx_list = glob.glob('/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/outcome_summary_stats/*.ma') # paths so we can list out names of diseases we have sum stats for   #<-- This line 'new' code as at 2023-12-19 19:05GMT (HD).\n",
    "#Note that as at 2023-12-21 16:55GMT (HD), according to Nautilus as on the Brutus this \"Folder\" (i.e., \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/outcome_summary_stats\") \"is Empty\" so it doesn't contain any *.ma files actually\n",
    "#\n",
    "#\n",
    "dx_clean = [] \n",
    "for x in dx_list:\n",
    "    if 'metaBrain' not in x:\n",
    "        dx_clean.append(x.split('/')[-1].rsplit('.')[0]) # strip for disease name\n",
    "dx_clean.sort() # sort list of diseases in alphabetical order\n",
    "dx_sum = dx_clean[1:] # cut off AD since we have different stats to use\n",
    "dx_sum38 = dx_sum.copy()\n",
    "#dx_sum.remove('LBD') # remove LBD to use different stats   #<-- TEMPorarily(?) commented this line as at 2023-11-24 23:01GMT (HD) as execution was giving an error:\n",
    "#====\n",
    "#---------------------------------------------------------------------------\n",
    "#ValueError                                Traceback (most recent call last)\n",
    "#Input In [8], in <cell line: 9>()\n",
    "#      7 dx_sum = dx_clean[1:] # cut off AD since we have different stats to use\n",
    "#      8 dx_sum38 = dx_sum.copy()\n",
    "#----> 9 dx_sum.remove('LBD') # remove LBD to use different stats\n",
    "#     10 dx_sum.remove('FTDold')\n",
    "#\n",
    "#ValueError: list.remove(x): x not in list\n",
    "#====\n",
    "#dx_sum.remove('FTDold') # remove old FTD sum stats   #<-- TEMPorarily(?) commented this line as at 2023-11-24 23:03GMT (HD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "e6529aaf-e96f-415a-9a57-1e6b932805a5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# sumstats gr37\n",
    "#\n",
    "#\n",
    "#\n",
    "#sumstats_pd = '/home/hammie/DATA/nalls2019_no23AndMe-20221121T130807Z-001/nalls2019_no23AndMe/nallsEtAl2019_excluding23andMe_allVariants.tab.zip'   #<-- This line 'new' code as at 2023-11-24 22:55GMT (HD) -- as at present I'm aware that for actual operationalization of this code, we'd most probably need to unzip the linked file first indeed. #Commented this line as at 2023-12-11 15:58GMT (HD).\n",
    "#\n",
    "#sumstats_pd = '\"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariants_mygwasDOTmaFormatFieldsOnly.tab\"'   #<-- This line 'new' code as at 2023-12-11 15:58GMT (HD) -- as at present I'm aware that for actual operationalization of this code, we'd most probably need to unzip the linked file first indeed -- also, keep fields as in \"https://yanglab.westlake.edu.cn/software/smr/#SMR\" example \"mygwas.ma\" only. Also, note that (as per \"https://yanglab.westlake.edu.cn/software/smr/#SMR\") \"The headers are not keywords and will be omitted by the program\". #Commented this line as at 2023-12-21 12:24GMT (HD).\n",
    "#\n",
    "sumstats_pd = '\"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\"'   #<-- This line 'new' code as at 2023-12-11 15:58GMT (HD) -- as at present I'm aware that for actual operationalization of this code, we'd most probably need to unzip the linked file first indeed -- also, keep fields as in \"https://yanglab.westlake.edu.cn/software/smr/#SMR\" example \"mygwas.ma\" only. Also, note that (as per \"https://yanglab.westlake.edu.cn/software/smr/#SMR\") \"The headers are not keywords and will be omitted by the program\". #<-- This line 'new' code as at 2023-12-21 12:26GMT (HD).\n",
    "#\n",
    "#\n",
    "### START Code TEMPorarily(?) commented as at 2023-11-24 23:04GMT (HD).\n",
    "#\n",
    "#sumstats_ad = '.../omicSynth/intermediate_results_AD_bellenguez/AD_hg19_smr.txt'\n",
    "#sumstats_bd = '.../omicSynth/intermediate_results_BD_schiz/BDI_hg19_smr.txt'\n",
    "#sumstats_lbd = '.../omicSynth/lbd_smr/LBD_37.ma'\n",
    "#\n",
    "### END Code TEMPorarily(?) commented as at 2023-11-24 23:04GMT (HD).\n",
    "#\n",
    "#\n",
    "sumstats = []\n",
    "for x in dx_sum:\n",
    "    sumstats.append(glob.glob(f'.../omicSynth/outcome_summary_stats/{x}.ma')[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "f379fe0f-e30b-4442-b256-5fa179139271",
   "metadata": {},
   "outputs": [],
   "source": [
    "# sumstats gr38\n",
    "#\n",
    "#\n",
    "#\n",
    "### START Code TEMPorarily(?) commented as at 2023-11-24 23:06GMT (HD).\n",
    "#\n",
    "#sumstats_meta_ad = '.../omicSynth/intermediate_results_AD_bellenguez/AD_metabrain_version_smr.txt'\n",
    "#sumstats_meta_bd = '.../omicSynth/intermediate_results_BD_schiz/BDI_hg38_smr.txt'\n",
    "#\n",
    "### END Code TEMPorarily(?) commented as at 2023-11-24 23:06GMT (HD).\n",
    "#\n",
    "#\n",
    "sumstats_meta = []\n",
    "for x in dx_sum38:\n",
    "    sumstats_meta.append(glob.glob(f'.../omicSynth/outcome_summary_stats/{x}_metaBrain_gr38_edition.ma')[0])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d1cbdb01-9304-4978-82d7-1e0a16548671",
   "metadata": {},
   "source": [
    "### GR37 Runs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "3ce04e2c-fa33-4011-bddb-a5d34c408cc9",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "# define remaining paths\n",
    "#\n",
    "#\n",
    "#out_dir = '.../omicSynth/v8/inter_test'   #<-- Commented this line as at 2023-11-24 23:13GMT (HD).\n",
    "#\n",
    "#\n",
    "out_dir = '/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test'   #<-- This line 'new' code as at 2023-11-24 23:15GMT (HD). #Commented this line as at 2023-12-21 11:17GMT (HD), as in OS Terminal as on Brutus I got the following Error: #... -- and uncommented this line again as at 2023-12-21 11:48GMT (HD).\n",
    "#====\n",
    "#(base) hammie@hammie-Default-string:/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr$ smr-1.3.1 --gwas-summary \\\"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariants_mygwasDOTmaFormatFieldsOnly.tab\\\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr22 --ld-upper-limit --out /media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr22 --bfile \\\"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\\\"\n",
    "#*******************************************************************\n",
    "#* Summary-data-based Mendelian Randomization (SMR)\n",
    "#* Version 1.3.1\n",
    "#* Build at Sep 21 2022 12:13:19, by GCC 8.3\n",
    "#* (C) 2015 Futao Zhang, Zhihong Zhu and Jian Yang\n",
    "#* The University of Queensland\n",
    "#* MIT License\n",
    "#*******************************************************************\n",
    "#Analysis started: 11:9:18,Thu Dec 21,2023\n",
    "#\n",
    "#Options:\n",
    "#--gwas-summary \"/media/hammie/My\n",
    "#\n",
    "#Error: can not open the file [\"/media/hammie/My] to read.\n",
    "#\n",
    "#Analysis completed: 11:9:18,Thu Dec 21,2023\n",
    "#Computational time: 0:0:0\n",
    "#(base) hammie@hammie-Default-string:/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr$ pwd\n",
    "#/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr\n",
    "#(base) hammie@hammie-Default-string:/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr$\n",
    "#====\n",
    "#\n",
    "#out_dir = '\"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test\"'   #<-- This line 'new' code as at 2023-12-21 11:17GMT (HD). #Commented this line as at 2023-12-21 11:47GMT (HD).\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "90e1c7ae-0121-40a7-a122-1197b8c94894",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cd \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/intermediate_results/\"\n",
      "Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr22 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr22\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n"
     ]
    }
   ],
   "source": [
    "# run SMR process for AD and BD seperate due to different summary stat paths\n",
    "#\n",
    "#\n",
    "SMR('PD',sumstats_pd, work_dir, out_dir)   #<-- This line 'new' code as at 2023-11-24 23:08GMT (HD)\n",
    "#\n",
    "#\n",
    "### START Code temporarily(?) commented as at 2023-12-08 15:15GMT (HD), to allow execution of \"run cell\" (i.e., \"Ctrl-Enter\") for the code line directly above within this same cell\n",
    "#\n",
    "#SMR('AD',sumstats_ad, work_dir, out_dir)\n",
    "#SMR('BD',sumstats_bd, work_dir, out_dir)\n",
    "#SMR('LBD',sumstats_lbd, work_dir, out_dir)\n",
    "#\n",
    "### END Code temporarily(?) commented as at 2023-12-08 15:15GMT (HD)\n",
    "\n",
    "#~--> As at 2023-12-08 15:24GMT (HD) led to \"error\":\n",
    "#\n",
    "#====\n",
    "#Executing: swarm -b 15 -f \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/swarms/PD.SMR_swarm\" -g 48 -t 12 --logdir \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/swarms\" --module SMR\n",
    "#\n",
    "#---------------------------------------------------------------------------\n",
    "#FileNotFoundError                         Traceback (most recent call last)\n",
    "#Input In [19], in <cell line: 4>()\n",
    "#      1 # run SMR process for AD and BD seperate due to different summary stat paths\n",
    "#      2 #\n",
    "#      3 #\n",
    "#----> 4 SMR('PD',sumstats_pd, work_dir, out_dir)\n",
    "#\n",
    "#Input In [12], in SMR(disease, stats_path, work_dir, out_dir)\n",
    "#     43 # swarm command\n",
    "#     44 #cmd = f'swarm -b 15 -f .../omicSynth/v8/swarms/{disease}.SMR_swarm -g 48 -t 12 --logdir .../omicSynth/v8/swarms --module SMR'   #<-- Commented this line as at 2023-11-23 20:49GMT (HD).\n",
    "#     45 #\n",
    "#     46 cmd = f'swarm -b 15 -f \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/swarms/{disease}.SMR_swarm\" -g 48 -t 12 --logdir \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/swarms\" --module SMR'   #<-- This line 'new' code as at 2023-11-23 20:49GMT (HD).\n",
    "#---> 48 shell_do(cmd)\n",
    "#\n",
    "#Input In [11], in shell_do(command, log, return_log, make_part)\n",
    "#      2 print(f'Executing: {(\" \").join(command.split())}', file=sys.stderr)\n",
    "#      4 if make_part == False:\n",
    "#----> 5     res = subprocess.run(command.split(), stdout=subprocess.PIPE)\n",
    "#      6 else:\n",
    "#      7     res = subprocess.run(command, shell=True, stdout=subprocess.PIPE)\n",
    "#\n",
    "#File ~/anaconda3/lib/python3.9/subprocess.py:505, in run(input, capture_output, timeout, check, *popenargs, **kwargs)\n",
    "#    502     kwargs['stdout'] = PIPE\n",
    "#    503     kwargs['stderr'] = PIPE\n",
    "#--> 505 with Popen(*popenargs, **kwargs) as process:\n",
    "#    506     try:\n",
    "#    507         stdout, stderr = process.communicate(input, timeout=timeout)\n",
    "#\n",
    "#File ~/anaconda3/lib/python3.9/subprocess.py:951, in Popen.__init__(self, args, bufsize, executable, stdin, stdout, stderr, preexec_fn, close_fds, shell, cwd, env, universal_newlines, startupinfo, creationflags, restore_signals, start_new_session, pass_fds, user, group, extra_groups, encoding, errors, text, umask)\n",
    "#    947         if self.text_mode:\n",
    "#    948             self.stderr = io.TextIOWrapper(self.stderr,\n",
    "#    949                     encoding=encoding, errors=errors)\n",
    "#--> 951     self._execute_child(args, executable, preexec_fn, close_fds,\n",
    "#    952                         pass_fds, cwd, env,\n",
    "#    953                         startupinfo, creationflags, shell,\n",
    "#    954                         p2cread, p2cwrite,\n",
    "#    955                         c2pread, c2pwrite,\n",
    "#    956                         errread, errwrite,\n",
    "#    957                         restore_signals,\n",
    "#    958                         gid, gids, uid, umask,\n",
    "#    959                         start_new_session)\n",
    "#    960 except:\n",
    "#    961     # Cleanup if the child failed starting.\n",
    "#    962     for f in filter(None, (self.stdin, self.stdout, self.stderr)):\n",
    "#\n",
    "#File ~/anaconda3/lib/python3.9/subprocess.py:1821, in Popen._execute_child(self, args, executable, preexec_fn, close_fds, pass_fds, cwd, env, startupinfo, creationflags, shell, p2cread, p2cwrite, c2pread, c2pwrite, errread, errwrite, restore_signals, gid, gids, uid, umask, start_new_session)\n",
    "#   1819     if errno_num != 0:\n",
    "#   1820         err_msg = os.strerror(errno_num)\n",
    "#-> 1821     raise child_exception_type(errno_num, err_msg, err_filename)\n",
    "#   1822 raise child_exception_type(err_msg)\n",
    "#\n",
    "#FileNotFoundError: [Errno 2] No such file or directory: 'swarm'\n",
    "#====\n",
    "\n",
    "#As at 2023-12-21 16:47GMT (HD) gave as screen output:\n",
    "#====\n",
    "#Executing: cd \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/intermediate_results/\"\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr1 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr1\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr2 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr2\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr3 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr3\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr4 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr4\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr5 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr5\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr6 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr6\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr7 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr7\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr8 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr8\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr9 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr9\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr10 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr10\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr11 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr11\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr12 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr12\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr13 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr13\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr14 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr14\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr15 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr15\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr16 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr16\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr17 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr17\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr18 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr18\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr19 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr19\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr20 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr20\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr21 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr21\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/BrainMeta_cis_eqtl_summary/BrainMeta_cis_eQTL_chr22 --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_brain_eMeta_SMR_chr22\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#===="
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "840b2008-e4ae-4513-8b33-db0cac877310",
   "metadata": {},
   "outputs": [],
   "source": [
    "#SMR('FTD_pottier', '.../omicSynth/outcome_summary_stats/FTD_Pottier37.ma', work_dir, out_dir)   #<-- (Temporarily?) commented this line as at 2023-12-21 15:28GMT (HD)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "3a4e5682-a816-46da-a9ad-bd8eed79e8d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "### START Code (temporarily?) commented as at 2023-12-21 15:30GMT (HD).\n",
    "#\n",
    "## run for all other dx\n",
    "#for dx, stat in zip(dx_sum, sumstats):\n",
    "#    SMR(dx,stat, work_dir, out_dir)\n",
    "#\n",
    "### END Code (temporarily?) commented as at 2023-12-21 15:30GMT (HD)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "678d2256-09d6-41a9-b1f0-0688edc7674e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cd \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/intermediate_results/\"\n",
      "Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary /home/hammie/DATA/SMR/GTEx_V8_cis_eqtl_summary_lite/eQTL_besd_lite/A --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_A_GTEx_SMR_allChrs\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Input \u001b[0;32mIn [55]\u001b[0m, in \u001b[0;36m<cell line: 9>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m#As at 2023-12-21 16:17GMT (HD) 'tot hier' -- continue w/ running code for GTEx (allegedly fully in SMR format already, as per Alavarado et al. 2023-medRxiv MS pp '14'--'15')\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \n\u001b[1;32m      3\u001b[0m \u001b[38;5;66;03m### START Code (temporarily?) commented as at 2023-12-21 15:30GMT (HD) (exclude non-\"PD\" [Nalls et al., Lancet Neurol 2019] diseases in this first instance).\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[38;5;66;03m#\u001b[39;00m\n\u001b[1;32m      8\u001b[0m \u001b[38;5;66;03m# Run SMR process for PD, AD and BD separately due to different summary stat paths #<-- This line 'new' comment as at 2023-12-21 17:02GMT (HD).\u001b[39;00m\n\u001b[0;32m----> 9\u001b[0m \u001b[43mSMR_gtex\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mPD\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43msumstats_pd\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mwork_dir\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mout_dir\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mgtex_short\u001b[49m\u001b[43m)\u001b[49m\n",
      "Input \u001b[0;32mIn [54]\u001b[0m, in \u001b[0;36mSMR_gtex\u001b[0;34m(disease, stats_path, work_dir, out_dir, gtex_list)\u001b[0m\n\u001b[1;32m    332\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m part \u001b[38;5;129;01min\u001b[39;00m gtex_list[\u001b[38;5;241m1\u001b[39m]:\n\u001b[1;32m    333\u001b[0m     \u001b[38;5;66;03m#f.write(f'smr --gwas-summary {stats_path} --beqtl-summary .../omicSynth/SMR_omics/eQTLs/GTEx_v8/eQTL_besd_lite/{part} --out {out_dir}/{disease}_expression_{part}_GTEx_SMR_allChrs --bfile .../omicSynth/additional_reference_data/1KG_and_refFlats/1kgenomes/1kg_eur_1pct_ref_panel  --smr-multi \\n')   #<-- Commented this line as at 2023-12-21 17:27GMT (HD).\u001b[39;00m\n\u001b[1;32m    334\u001b[0m     \n\u001b[1;32m    335\u001b[0m     \u001b[38;5;66;03m### START code amended from SMR() function definition above as at 2023-12-21 17:29GMT (HD).\u001b[39;00m\n\u001b[1;32m    336\u001b[0m     \u001b[38;5;66;03m#\u001b[39;00m\n\u001b[1;32m    337\u001b[0m     \u001b[38;5;66;03m#\u001b[39;00m\n\u001b[1;32m    338\u001b[0m     cmd \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mPATH=$PATH:\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m; smr-1.3.1 --gwas-summary \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mstats_path\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m --beqtl-summary /home/hammie/DATA/SMR/GTEx_V8_cis_eqtl_summary_lite/eQTL_besd_lite/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mpart\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m --out \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mout_dir\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m/\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mdisease\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m_expression_\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mpart\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m_GTEx_SMR_allChrs\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m --bfile \u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m --smr-multi\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m--> 340\u001b[0m     \u001b[43mshell_do\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcmd\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmake_part\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\n",
      "Input \u001b[0;32mIn [2]\u001b[0m, in \u001b[0;36mshell_do\u001b[0;34m(command, log, return_log, make_part)\u001b[0m\n\u001b[1;32m      5\u001b[0m     res \u001b[38;5;241m=\u001b[39m subprocess\u001b[38;5;241m.\u001b[39mrun(command\u001b[38;5;241m.\u001b[39msplit(), stdout\u001b[38;5;241m=\u001b[39msubprocess\u001b[38;5;241m.\u001b[39mPIPE)\n\u001b[1;32m      6\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m----> 7\u001b[0m     res \u001b[38;5;241m=\u001b[39m \u001b[43msubprocess\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrun\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcommand\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mshell\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstdout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msubprocess\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mPIPE\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.9/subprocess.py:507\u001b[0m, in \u001b[0;36mrun\u001b[0;34m(input, capture_output, timeout, check, *popenargs, **kwargs)\u001b[0m\n\u001b[1;32m    505\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m Popen(\u001b[38;5;241m*\u001b[39mpopenargs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs) \u001b[38;5;28;01mas\u001b[39;00m process:\n\u001b[1;32m    506\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 507\u001b[0m         stdout, stderr \u001b[38;5;241m=\u001b[39m \u001b[43mprocess\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcommunicate\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43minput\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    508\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m TimeoutExpired \u001b[38;5;28;01mas\u001b[39;00m exc:\n\u001b[1;32m    509\u001b[0m         process\u001b[38;5;241m.\u001b[39mkill()\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.9/subprocess.py:1121\u001b[0m, in \u001b[0;36mPopen.communicate\u001b[0;34m(self, input, timeout)\u001b[0m\n\u001b[1;32m   1119\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_stdin_write(\u001b[38;5;28minput\u001b[39m)\n\u001b[1;32m   1120\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstdout:\n\u001b[0;32m-> 1121\u001b[0m     stdout \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mstdout\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1122\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstdout\u001b[38;5;241m.\u001b[39mclose()\n\u001b[1;32m   1123\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mstderr:\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "### START Code (temporarily?) commented as at 2023-12-21 15:30GMT (HD) (exclude non-\"PD\" [Nalls et al., Lancet Neurol 2019] diseases in this first instance).\n",
    "#\n",
    "#\n",
    "## run SMR process for AD and BD seperate due to different summary stat paths #<-- 'Commented out this comment line' as at 2023-12-21 17:01GMT (HD).\n",
    "#\n",
    "# Run SMR process for PD, AD and BD separately due to different summary stat paths #<-- This line 'new' comment as at 2023-12-21 17:02GMT (HD).\n",
    "SMR_gtex('PD',sumstats_pd, work_dir, out_dir, gtex_short)   #<-- This line 'new' code as at 2023-12-21 17:04GMT (HD).\n",
    "#\n",
    "#SMR_gtex('AD',sumstats_ad, work_dir, out_dir, gtex_short)\n",
    "#SMR_gtex('BD',sumstats_bd, work_dir, out_dir, gtex_short)\n",
    "#\n",
    "#\n",
    "### END Code (temporarily?) commented as at 2023-12-21 15:30GMT (HD)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "99b0e843-f4c8-4062-afa4-fb9a73957b7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "### START Code (temporarily?) commented as at 2023-12-22 22:28GMT (HD): exclude non-\"PD\" (Nalls et al., Lancet Neurol 2019) diseases in this first instance.\n",
    "#\n",
    "## run for all other dx\n",
    "#for dx, stat in zip(dx_sum, sumstats):\n",
    "#    SMR_gtex(dx,stat, work_dir, out_dir, gtex_short)\n",
    "#\n",
    "### END Code (temporarily?) commented as at 2023-12-22 22:28GMT (HD)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "779f62bb-a3e7-4303-9636-7a6e331f9cd2",
   "metadata": {},
   "outputs": [],
   "source": [
    "#SMR_gtex('FTD_pottier', '.../omicSynth/outcome_summary_stats/FTD_Pottier37.ma', work_dir, out_dir, gtex_short)   #<-- (Temporarily?) commented this line as at 2023-12-21 15:37GMT (HD): exclude non-\"PD\" (Nalls et al., Lancet Neurol 2019) diseases in this first instance."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "30e4ec5d-5e9d-4b27-98db-f5d2806e4e5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "### START Code (temporarily?) commented as at 2023-12-21 15:38GMT (HD): Consider only eQTLGen & GTEx as exposures in the first instance.\n",
    "#\n",
    "## run SMR process for AD and BD seperate due to different summary stat paths\n",
    "#SMR_psych('AD',sumstats_ad, work_dir, out_dir)\n",
    "#SMR_psych('BD',sumstats_bd, work_dir, out_dir)\n",
    "#\n",
    "### END Code (temporarily?) commented as at 2023-12-21 15:38GMT (HD)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9416b588-c65b-4ccd-a4d4-afdc410ae165",
   "metadata": {},
   "outputs": [],
   "source": [
    "#SMR_psych('LBD',sumstats_lbd, work_dir, out_dir)   #(Temporarily?) commented this line as at 2023-12-21 16:03GMT (HD): Consider only eQTLGen & GTEx as exposures in the first instance."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "698b432b-231f-4389-bd1f-7f9657f58f77",
   "metadata": {},
   "outputs": [],
   "source": [
    "#SMR_psych('PSP', sumstats[-4], work_dir, out_dir)   #(Temporarily?) commented this line as at 2023-12-21 16:03GMT (HD): Consider only eQTLGen & GTEx as exposures in the first instance."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "719acc88-7d26-458a-8e01-d85803e28f2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#SMR_psych('FTD_pottier', '.../omicSynth/outcome_summary_stats/FTD_Pottier37.ma', work_dir, out_dir)   #(Temporarily?) commented this line as at 2023-12-21 16:03GMT (HD): Consider only eQTLGen & GTEx as exposures in the first instance."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "af368338-d9b8-4435-a46d-836e2fce3949",
   "metadata": {},
   "outputs": [],
   "source": [
    "### START Code (temporarily?) commented as at 2023-12-21 16:04GMT (HD): Consider only eQTLGen & GTEx as exposures in the first instance.\n",
    "#\n",
    "# run for all other dx\n",
    "#for dx, stat in zip(dx_sum, sumstats):\n",
    "#    SMR_psych(dx,stat, work_dir, out_dir)\n",
    "#\n",
    "### END Code (temporarily?) commented as at 2023-12-21 16:04GMT (HD)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "83bb6690-9558-4519-8093-fd183e37bdb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "### START Code (temporarily?) commented as at 2023-12-21 16:05GMT (HD): Consider only eQTLGen & GTEx as exposures in the first instance.\n",
    "#\n",
    "## run SMR process for AD and BD seperate due to different summary stat paths\n",
    "#SMR_pQTL('AD',sumstats_ad, work_dir, out_dir)\n",
    "#SMR_pQTL('BD',sumstats_bd, work_dir, out_dir)\n",
    "#SMR_pQTL('LBD',sumstats_lbd, work_dir, out_dir)\n",
    "#\n",
    "### END Code (temporarily?) commented as at 2023-12-21 16:05GMT (HD)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "19506e6a-ecfc-4f54-9f46-9cdcceea6517",
   "metadata": {},
   "outputs": [],
   "source": [
    "### START Code (temporarily?) commented as at 2023-12-21 16:07GMT (HD): Consider only eQTLGen & GTEx as exposures in the first instance.\n",
    "#\n",
    "#SMR_pQTL('PSP', sumstats[-4], work_dir, out_dir)\n",
    "#\n",
    "#SMR_pQTL('FT2', '.../omicSynth/outcome_summary_stats/FTD_Pottier37.ma', work_dir, out_dir)\n",
    "#\n",
    "### END Code (temporarily?) commented as at 2023-12-21 16:07GMT (HD)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dc045f58-9d43-4f90-8f59-c2fefc15e41f",
   "metadata": {},
   "outputs": [],
   "source": [
    "### START Code (temporarily?) commented as at 2023-12-21 16:07GMT (HD): Consider only eQTLGen & GTEx as exposures in the first instance.\n",
    "#\n",
    "## run for all other dx\n",
    "#for dx, stat in zip(dx_sum, sumstats):\n",
    "#    SMR_pQTL(dx,stat, work_dir, out_dir)\n",
    "#\n",
    "### END Code (temporarily?) commented as at 2023-12-21 16:07GMT (HD)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "032fb216-9c5f-43d0-8ee0-a01656f89cc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "### START Code (temporarily?) commented as at 2023-12-21 16:08GMT (HD): Consider only eQTLGen & GTEx as exposures in the first instance.\n",
    "#\n",
    "## run SMR process for AD and BD seperate due to different summary stat paths\n",
    "#SMR_eQTL('AD',sumstats_ad, work_dir, out_dir)\n",
    "#SMR_eQTL('BD',sumstats_bd, work_dir, out_dir)\n",
    "#SMR_eQTL('LBD',sumstats_lbd, work_dir, out_dir)\n",
    "#\n",
    "### END Code (temporarily?) commented as at 2023-12-21 16:08GMT (HD)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0298f58f-5e5d-4955-bf43-e3291cb1c6aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "#SMR_eQTL('FTD_pottier', '.../omicSynth/outcome_summary_stats/FTD_Pottier37.ma', work_dir, out_dir)   #(Temporarily?) commented this line as at 2023-12-21 16:09GMT (HD): Consider only eQTLGen & GTEx as exposures in the first instance."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "caaf92a9-a182-4b40-b52f-fa01e451e4d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "### START Code (temporarily?) commented as at 2023-12-21 16:09GMT (HD): Consider only eQTLGen & GTEx as exposures in the first instance.\n",
    "#\n",
    "## run for all other dx\n",
    "#for dx, stat in zip(dx_sum, sumstats):\n",
    "#    SMR_eQTL(dx,stat, work_dir, out_dir)\n",
    "#\n",
    "### END Code (temporarily?) commented as at 2023-12-21 16:09GMT (HD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "8e9e8209-709b-4905-8707-969aeb27e1a2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: cd \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/intermediate_results/\"\n",
      "Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary \"/media/hammie/My Passport/DATA/eQTLGen/cis-eQTLs-full_eQTLGen_AF_incl_nr_formatted_20191212.new.txt_besd-dense\" --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_blood_eQTLgen_SMR_allChrs\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n"
     ]
    }
   ],
   "source": [
    "### START Code (temporarily?) commented as at 2023-12-22 22:35GMT (HD): exclude non-\"PD\" (Nalls et al., Lancet Neurol 2019) diseases in this first instance.\n",
    "#\n",
    "#\n",
    "## run SMR process for AD and BD seperate due to different summary stat paths #<-- 'Commented out this comment line' as at 2023-12-22 22:37GMT (HD).\n",
    "#\n",
    "# Run SMR process for AD; BD; LBD; and PD separately due to different summary stat paths #<-- This line 'new' comment as at 2023-12-22 22:38GMT (HD).\n",
    "#SMR_eqtlGen('AD',sumstats_ad, work_dir, out_dir)\n",
    "#SMR_eqtlGen('BD',sumstats_bd, work_dir, out_dir)\n",
    "#SMR_eqtlGen('LBD',sumstats_lbd, work_dir, out_dir)\n",
    "#\n",
    "SMR_eqtlGen('PD',sumstats_pd, work_dir, out_dir)   #<-- This line 'new' code as at 2023-12-22 22:36GMT (HD).\n",
    "#--> As at 2023-12-22 22:54GMT (HD) this yielded the following screen output here:\n",
    "#====\n",
    "#Executing: cd \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/intermediate_results/\"\n",
    "#Executing: PATH=$PATH:\"/media/hammie/My Passport/SOFTWARE/smr-1.3.1-linux-x86_64\"; smr-1.3.1 --gwas-summary \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/pd_smr/nallsEtAl2019_excluding23andMe_allVariantsASrsIDs_mygwasDOTmaFormatFieldsOnly.tab\" --beqtl-summary \"/media/hammie/My Passport/DATA/eQTLGen/cis-eQTLs-full_eQTLGen_AF_incl_nr_formatted_20191212.new.txt_besd-dense\" --out \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/v8/inter_test/PD_expression_blood_eQTLgen_SMR_allChrs\" --bfile \"/media/hammie/My Passport/SOFTWARE/omicsynthcustomanalysis/additional_reference_data/1KG_and_refFlats/1kgenomes/EUR\" --smr-multi\n",
    "#====\n",
    "#\n",
    "#\n",
    "### END Code (temporarily?) commented as at 2023-12-22 22:35GMT (HD)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4457371d-6fbb-45d6-aa1e-8246b25f4fb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "### START Code (temporarily?) commented as at 2023-12-22 22:38GMT (HD): exclude non-\"PD\" (Nalls et al., Lancet Neurol 2019) diseases in this first instance.\n",
    "#\n",
    "## run for all other dx\n",
    "#for dx, stat in zip(dx_sum, sumstats):\n",
    "#    SMR_eqtlGen(dx,stat, work_dir, out_dir)\n",
    "#\n",
    "### END Code (temporarily?) commented as at 2023-12-22 22:38GMT (HD)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d38a2deb-07f1-4121-a171-0932cf48a322",
   "metadata": {},
   "source": [
    "### MetaBrain GR38 Runs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "55bd24ee-c699-4bc7-bb30-e94df86a676e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#\"meta\" as for/with \"metaBrain\" data indeed -- REF SMR_meta() function definition above as at 2023-12-21 16:13GMT.\n",
    "\n",
    "### START Code (temporarily?) commented as at 2023-12-21 16:14GMT (HD): Consider only eQTLGen & GTEx as exposures in the first instance.\n",
    "#\n",
    "## run SMR process for AD and BD seperate due to different summary stat paths\n",
    "#SMR_meta('AD',sumstats_meta_ad, work_dir, out_dir)\n",
    "#SMR_meta('BD',sumstats_meta_bd, work_dir, out_dir)\n",
    "#\n",
    "### END Code (temporarily?) commented as at 2023-12-21 16:14GMT (HD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "848bdca5-8289-42d9-a586-4286c03c3a83",
   "metadata": {},
   "outputs": [],
   "source": [
    "#SMR_meta('FTD_pottier', '.../omicSynth/outcome_summary_stats/FTD_Pottier37_metaBrain_gr38_edition.ma', work_dir, out_dir)   #(Temporarily?) commented this line as at 2023-12-21 16:15GMT (HD): Consider only eQTLGen & GTEx as exposures in the first instance."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "881e26f0-30de-46de-9c28-9855596d6306",
   "metadata": {},
   "outputs": [],
   "source": [
    "### START Code (temporarily?) commented as at 2023-12-21 16:16GMT (HD): Consider only eQTLGen & GTEx as exposures in the first instance.\n",
    "#\n",
    "## g38\n",
    "#for dx, stat in zip(dx_sum, sumstats_meta):\n",
    "#    SMR_meta(dx,stat, work_dir, out_dir)\n",
    "#\n",
    "### END Code (temporarily?) commented as at 2023-12-21 16:16GMT (HD)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
